org.opent2t.sample.binaryswitch.superpopular/com.insteon.binaryswitch/js/thingTranslator.js

```diff
@@ -0,0 +1,174 @@
+'use strict';
+var crypto = require('crypto');
+
+// This code uses ES2015 syntax that requires at least Node.js v4.
+// For Node.js ES2015 support details, reference http://node.green/
+
+function validateArgumentType(arg, argName, expectedType) {
+    if (typeof arg === 'undefined') {
+        throw new Error('Missing argument: ' + argName + '. ' +
+            'Expected type: ' + expectedType + '.');
+    } else if (typeof arg !== expectedType) {
+        throw new Error('Invalid argument: ' + argName + '. ' +
+            'Expected type: ' + expectedType + ', got: ' + (typeof arg));
+    }
+}
+
+/**
+ * Finds a resource for an entity in a schema
+ */
+function findResource(schema, di, resourceId) {
+    // Find the entity by the unique di
+    var entity = schema.entities.find((d) => {
+        return d.di === di;
+    });
+
+    if (!entity) {
+        throw new Error('NotFound');
+    }
+
+    var resource = entity.resources.find((r) => {
+        return r.id === resourceId;
+    });
+
+    if (!resource) throw new Error('NotFound');
+    return resource;
+}
+
+/**
+ * Generate a GUID for given an ID.
```

Comment:
Fixed.

---
In reply to: [98084552](https://github.com/openT2T/translators/pull/149#discussion_r98084552) [](ancestors = 98084552)